api/hasura/auth.ts

```diff
@@ -19,9 +19,15 @@ export default async function handler(req, res) {
       },
     });
     assert(tokenRow, 'The token provided was not recognized');
+    const profile = await prisma.profile.findFirst({
```

Comment:
Try using a [relation filter](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#relation-filters) to combine this and the above into a single query